cmd/handler-api.go

```diff
@@ -48,6 +51,40 @@ type apiConfig struct {
 	deleteCleanupInterval       time.Duration
 }
 
+const cgroupLimitFile = "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+
+func availableMemory() (available uint64) {
+	available = 8 << 30 // Default to 8 GiB when we can't find the limits.
+
+	if runtime.GOOS == "linux" {
+		buf, err := ioutil.ReadFile(cgroupLimitFile)
+		if err != nil {
+			return
+		}
+		cgroupLimit, err := strconv.ParseUint(string(buf), 10, 64)
+		if err != nil {
+			return 0
+		}
+		available = cgroupLimit
+	}
+
+	// No limit set, It's the highest positive signed 64-bit
+	// integer (2^63-1), rounded down to multiples of 4096 (2^12),
+	// the most common page size on x86 systems - for cgroup_limits.
+	if available != 9223372036854771712 {
+		// This means cgroup memory limit is configured.
+		return
```

Comment:
We are returning early here with available = 8 Gb when the OS is not 'linux'

Reply:
Yeah. move this into the linux section.